src/pydargs/__init__.py

```diff
@@ -22,14 +22,48 @@ def parse(tp: Type[Dataclass], args: Optional[list[str]] = None) -> Dataclass:
 def _create_parser(tp: Type[Dataclass]) -> ArgumentParser:
     parser = ArgumentParser()
     for field in fields(tp):
-        parser.add_argument(
-            f"--{field.name.replace('_', '-')}",
-            default=field.default,
-            dest=field.name,
-            help=f"Override field {field.name}.",
-            type=field.type,
-        )
+        if field.type is bool:
+            if arg_type := field.metadata.get("bool_arg_type", None):
```

Comment:
Wouldn't it be nicer to have the flags as default?

I would certainly prefer a metadata field that is a bit more self-explanatory. E.g. `as_flags=True`

Reply:
Maybe `add_flags` even?